providers/implementations/kem/ec_kem.c

```diff
@@ -405,10 +405,10 @@ int ossl_ec_dhkem_derive_private(EC_KEY *ec, BIGNUM *priv,
         return 0;
 
     /* ikmlen should have a length of at least Nsk */
-    if (ikmlen < info->Nsecret) {
+    if (ikmlen < info->Nsk) {
```

Comment:
Looking at RFC9180, we see:

* Nsk: The length in bytes of an encoded private key for this KEM.
* Nsecret: The length in bytes of a KEM shared secret produced by this KEM.

The seed `ikm` should have sufficient entropy to ensure that the derived shared secret is not weak, but this is NOT a function of the private key size (which is an internal detail of the algorithm), rather it should scale with the shared secret size (the security of the algorithm). So I think that what was wrong, was the comment, and NOT the code.

It think this should be `Nsecret`, and the *comment* should then be fixed.

Reply:
I agree with that.

Reply:
That said, RFC9180, [Section 7.1.3](https://www.rfc-editor.org/rfc/rfc9180#derive-key-pair), clearly disagrees with my intuition, sorry about that. I'm approving the PR.

Reply:
Yeah

Reply:
Glad, thanks! LGTM